lib/active_model/serializer/associations.rb

```diff
@@ -74,7 +74,8 @@ def has_one(name, options = {}, &block) # rubocop:disable Style/PredicateName
         # @api private
         #
         def associate(reflection)
-          self._reflections << reflection
+          key = reflection.options[:key]
+          key ? self._reflections[key] = reflection : self._reflections[reflection.name] = reflection
```

Comment:
feel free to open a new PR, as this one is already closed. (It'll also keep the review process cleaner, too)

But yeah, basically:
 - short lines
 - each line should do only one thing